### PR TITLE
docs: containize instructions with `docker build`

### PR DESCRIPTION
**Pull request by aarnphm**
Mention inside our migration guides about buildkit changes

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: docs/source/guides/migration.rst [hunk 1]
```diff
@@ -2,9 +2,9 @@
 1.0 Migration Guide
 ===================
 
-BentoML version 1.0.0 APIs are backward incompatible with version 0.13.1. However, most of the common 
-functionality can be achieved with the new version. We will guide and demonstrate the migration by 
-transforming the `quickstart <https://github.com/bentoml/BentoML/tree/main/examples/quickstart>`_ gallery project 
+BentoML version 1.0.0 APIs are backward incompatible with version 0.13.1. However, most of the common
+functionality can be achieved with the new version. We will guide and demonstrate the migration by
+transforming the `quickstart <https://github.com/bentoml/BentoML/tree/main/examples/quickstart>`_ gallery project
```
Comment (sauyon) [inline docs/source/guides/migration.rst:7]: Will probably need a reflow. (I know this is a just a whitespace change, but we may as well make these changes)

```suggestion
BentoML version 1.0.0 APIs are backward incompatible with version 0.13.1. However, most of the
functionality is maintained in the new version. We will demonstrate this by
migrating the `quickstart <https://github.com/bentoml/BentoML/tree/main/examples/quickstart>`_ gallery project.
```

File: docs/source/guides/migration.rst [hunk 2]
```diff
@@ -68,47 +68,46 @@ Next, we will transform the service definition module and breakdown each section
...
+          iris_clf_runner = bentoml.sklearn.get("iris_clf:latest").to_runner()
 
-            svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
+          svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
 
-            @svc.api(input=PandasDataFrame(), output=NumpyNdarray())
-            def predict(input_series: pd.DataFrame) -> np.ndarray:
-                result = iris_clf_runner.predict.run(input_series)
-                return result
+          @svc.api(input=bentoml.io.PandasDataFrame(), output=bentoml.io.NumpyNdarray())
```
Comment (sauyon) [inline docs/source/guides/migration.rst:107]: Why this change? It's not in the actual example.

File: docs/source/guides/migration.rst [hunk 3]
```diff
@@ -328,6 +328,25 @@ You can run the docker image to start the service.
     2022-07-01T21:57:48+0000 [INFO] [api_server:3] Service loaded from Bento directory: bentoml.Service(tag="iris_classifier:6otbsmxzq6lwbgxi", path="/home/bentoml/bento/")
     2022-07-01T21:57:48+0000 [INFO] [api_server:4] Service loaded from Bento directory: bentoml.Service(tag="iris_classifier:6otbsmxzq6lwbgxi", path="/home/bentoml/bento/")
 
+.. note::
+
+   BentoML 1.x leverage BuildKit <https://github.com/moby/buildkit>`_, to
```
Comment (sauyon) [inline docs/source/guides/migration.rst:333]:
```suggestion
   BentoML 1.x leverages BuildKit <https://github.com/moby/buildkit>`_, to
```

File: docs/source/guides/migration.rst [hunk 4]
```diff
@@ -328,6 +328,25 @@ You can run the docker image to start the service.
...
+   containerize Bentos.
+
+   BuildKit has been `integrated into Docker 18.09 and higher <https://docs.docker.com/develop/develop-images/build_enhancements/>`_.
+   Providing the environment variable ``DOCKER_BUILDKIT=1`` will enable BuildKit:
+
+   .. code-block:: bash
+
+      $ DOCKER_BUILDKIT=1 bentoml containerize iris_classifier:latest
```
Comment (sauyon) [inline docs/source/guides/migration.rst:341]: Can we automatically set this in our CI and not require `buildx`?

File: docs/source/guides/migration.rst [hunk 5]
```diff
@@ -328,6 +328,25 @@ You can run the docker image to start the service.
...
+
+.. seealso::
+
+   :ref:`Containerize Bentos <concepts/bento:Docker Options>` to learn more about our
+   docker options and how to customize the docker image. :ref:`guides/containerization:Advanced Containerization` goes into more 
+   details about advanced containerization features that BentoML provides, for those who
+   are interested.
```
Comment (sauyon) [inline docs/source/guides/migration.rst:348]:
```suggestion
   :ref:`concepts/bento:Docker Options` describes our
   docker options and how to customize the docker image.
   :ref:`guides/containerization:Advanced Containerization` goes into more 
   details about advanced containerization features that BentoML provides.
```

File: docs/source/guides/migration.rst (hunk 2 above)
Comment (aarnphm) [reply docs/source/guides/migration.rst:107]: let me change this back

File: docs/source/guides/migration.rst (hunk 4 above)
Comment (aarnphm) [reply docs/source/guides/migration.rst:342]: Our containerize already set this environment variable.

File: docs/source/guides/migration.rst [hunk 6]
```diff
@@ -328,6 +329,45 @@ You can run the docker image to start the service.
...
+
+
+.. seealso::
+
+   :ref:`Containerize Bentos <concepts/bento:Docker Options>` to learn more about our
+   docker options and how to customize the docker image.
+
+   :ref:`guides/containerization:Advanced Containerization` goes into more 
+   details about advanced containerization features that BentoML provides, for those who
+   are interested.
```
Comment (aarnphm) [inline docs/source/guides/migration.rst:369]:
```suggestion
   :ref:`Containerize Bentos <concepts/bento:Docker Options>` describes our
   docker options and how to customize the docker image.

   :ref:`guides/containerization:Advanced Containerization` goes into more 
   details about the advanced containerization features that BentoML provides.
```
